File: src/test/java/discretemath/structure/function/StaticFunctionTest.java
package discretemath.structure.function;

import java.util.Map;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class StaticFunctionTest {

	@Test
	void isOneToOne() {
		StaticFunction<Integer, Integer> function = StaticFunction.from(Map.of(1, 2, 4, 5, 6, 7));

		assertTrue(function.isOneToOne());

		StaticFunction<Integer, Integer> function2 = StaticFunction.from(Map.of(1, 2, 4, 2, 6, 7));

		assertFalse(function2.isOneToOne());
	}

	@Test
	void getInverse() {
		StaticFunction<Integer, Integer> function = StaticFunction.from(Map.of(1, 5, 2, 8, 3, 10));

		Function<Integer, Integer> inverseFunction = function.getInverse();

		assertEquals(1, inverseFunction.apply(5));
		assertEquals(2, inverseFunction.apply(8));
		assertEquals(3, inverseFunction.apply(10));
	}
}

File: src/test/scala/discretemath/test/structure/function/StaticFunctionTest.scala
package discretemath.test.structure.function

import discretemath.structure.function.StaticFunction
import org.scalatest.funsuite.AnyFunSuite

class StaticFunctionTest extends AnyFunSuite {

  test("isOneToOne") {
    val function = StaticFunction(Map(1 -> 2, 4 -> 5, 6 -> 7));

    assert(function.isOneToOne);

    val function2 = StaticFunction(Map(1 -> 2, 4 -> 2, 6 -> 7));

    assert(!function2.isOneToOne);
  }

  test("getInverse") {
    val function = StaticFunction(Map(1 -> 5, 2 -> 8, 3 -> 10));

    val inverseFunction = function.getInverse;

    assert(inverseFunction.apply(5) == 1);
    assert(inverseFunction.apply(8) == 2);
    assert(inverseFunction.apply(10) == 3);
  }
}